index.js

```diff
@@ -3,26 +3,39 @@ const {createFilter} = require("rollup-pluginutils");
 
 const importToGlobals = require("./lib/import-to-globals");
 
-function createPlugin(globals, {include, exclude} = {}) {
+function createPlugin(globals, {include, exclude, dynamicWrapper} = {}) {
   const filter = createFilter(include, exclude);
   return {
     name: "rollup-plugin-external-globals",
     transform
   };
-  
+
   function transform(code, id) {
-    if (id[0] !== "\0" && !filter(id)) {
+    if (!globals || (id[0] !== "\0" && !filter(id))) {
       return;
     }
-    if (Object.keys(globals).every(id => !code.includes(id))) {
-      return;
+    let getName;
+    if (typeof globals === "object") {
+      if (Object.keys(globals).every(id => !code.includes(id))) {
+        return;
+      }
+      getName = function (name) {
+        if (globals.hasOwnProperty(name)) {
+          return globals[name];
+        }
+      };
+    } else if (globals instanceof Function) {
+      getName = globals;
+    } else {
+      return false;
```

Comment:
We can throw an error here:
```js
throw new TypeError(`Unexpected type of 'globals', got '${typeof globals}'`);
```
It will also be a good idea to lift this check to the initialization phase.